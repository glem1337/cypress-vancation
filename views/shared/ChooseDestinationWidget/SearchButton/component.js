import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'antd';

import useButtonGradient from 'utils/hooks/useButtonGradient';

const SearchButton = ({ onClick }) => {
  const { containerRef } = useButtonGradient();

  return (
    <Button
      className="main-btn--gradient ant-btn-icon-only d-none d-md-inline-block choose-destination__search_btn"
      size="large"
      shape="circle"
      icon={<i className="icon icon-search" />}
      onClick={onClick}
      ref={containerRef}
    >
      <span className="main-btn__gradient-inner" />
    </Button>
  );
};

SearchButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default SearchButton;
