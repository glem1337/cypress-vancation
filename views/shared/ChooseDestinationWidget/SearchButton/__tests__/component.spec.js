import React from 'react';
import { shallow } from 'enzyme';

import SearchButton from '../component';

const mockedHookData = {
  containerRef: {},
};
jest.mock('utils/hooks/useButtonGradient', () => jest.fn(() => mockedHookData));

describe('SearchButton component tests', () => {
  const props = {
    onClick: jest.fn(),
  };

  const component = shallow(<SearchButton {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
