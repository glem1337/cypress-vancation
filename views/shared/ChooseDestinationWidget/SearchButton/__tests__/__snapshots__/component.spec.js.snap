// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton component tests matches snapshot 1`] = `
<Button
  className="main-btn--gradient ant-btn-icon-only d-none d-md-inline-block choose-destination__search_btn"
  icon={
    <i
      className="icon icon-search"
    />
  }
  onClick={[MockFunction]}
  shape="circle"
  size="large"
>
  <span
    className="main-btn__gradient-inner"
  />
</Button>
`;
